rsc/src/main/java/com/cloudera/livy/rsc/rpc/RpcServer.java

```diff
@@ -67,11 +68,35 @@
   private final ConcurrentMap<String, ClientInfo> pendingClients;
   private final RSCConf config;
 
-  public RpcServer(RSCConf lconf) throws IOException, InterruptedException {
+  @VisibleForTesting
+  RpcServer(RSCConf lconf) throws IOException, InterruptedException {
+    this(lconf, false);
+  }
+
+  public RpcServer(RSCConf lconf, boolean isLauncher) throws IOException, InterruptedException {
```

Comment:
The code can be simplified a lot if we pass `int port` to the constructor instead of `boolean isLauncher`.